card/mod/machines/lib/javascript/decko/filter.js.coffee

```diff
@@ -1,187 +1,155 @@
-$.extend decko,
-  filterCategorySelected: ($selected_item) ->
-    addFilterDropdown = $selected_item.closest("._add-filter-dropdown")
-    category = $selected_item.data("category")
-    widget = addFilterDropdown.closest("._filter-widget")
-    removeCategoryOption(addFilterDropdown, category)
-    showFilterInputField(category, widget)
+decko.slotReady (slot) ->
+  slot.find("._filter-widget").each ->
+    if slot[0] == $(this).slot()[0]
```

Comment:
THIS is the reason why I want some API like "slot loaded".

If `div._filter-widget` is in a slot within a slot, then I have to do something like this to prevent the code from being called twice.

I suspect there are already many cases where we are getting repeat calls for this.  In this case it breaks if there are repeat calls, but even if it doesn't break, it's wasteful.

Reply:
It's easy to get the closest slot if needed. If you check all slotReady calls I'm fine with changing it so that it is only triggered for the root slot and not for nested slots.

Reply:
When you say "it's easy to get the closest slot", you're talking about the outermost slot(s) in any given call, right?  (As opposed to "closest" in the jquery sense, meaning there's some better way to do what I've done here)